src/components/Editor/CadenceEditor/ControlPanel/index.tsx

```diff
@@ -284,17 +285,17 @@ const ControlPanel: React.FC<ControlPanelProps> = (props) => {
         }
 
         case EntityType.ContractTemplate: {
-          // TODO: Deploy to selected account after we have the deployment account selector
           if (
             accounts[active.index] &&
-            accounts[active.index].deployedContracts.length > 0 &&
+            active.index in
+              (accounts[selectedAccounts[0] || 0]?.deployedContracts || []) &&
             !showPrompt
           ) {
             setProcessingStatus(false);
             setShowPrompt(true);
             return;
           }
-          const selectedAccountId = selected[0] || 0;
+          const selectedAccountId = selectedAccounts[0] || 0;
```

Comment:
abstract this logic into a helper function, the same snippet is used in four places

Reply:
created a helper for other instances.